File: twitter-nwt/Models/Comment.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace twitter_nwt.Models
{
    public class Comment
    {
        [Key]
        public int Id { get; set; }

        public string Text { get; set; }

        [JsonIgnore]
        public User User { get; set; }

        [JsonIgnore]
        public Tweet Tweet { get; set; }
    }
}

File: twitter-nwt/Assets/app/Common/Models/Comment.ts
import { User } from './User';
import { Tweet } from './Tweet';

export class Comment {
    public id: number;
    public text: string;
    public user: User;
    public tweet: Tweet;

    constructor(id: number = 1, text: string, user: User, tweet: Tweet) {
        this.id = id;
        this.text = text;
        this.user = user;
        this.tweet = tweet;
    }
}